board/piksiv3/rootfs-overlay/etc/init.d/copy_sys_logs.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/ash
+
+export LOGDIR="/media/mmcblk0p1/logs"
+export LOGLEVEL="--info"
+_dir_wait () {
+    [[ $# -lt 3 ]] && {
+        echo "Usage: ${FUNCNAME} <num_timeout> <len_timeout_ms> <path>"; return 1
+    }
+    RET=1
+    REPEATS="${1}"
+    LEN_TIMEOUT_MS="${2}"
+    MNT_PATH="${3}"
+    until [ -x $MNT_PATH ] || [ $REPEATS -lt 1 ]; do
+       usleep $(( $LEN_TIMEOUT_MS * 1000 ))
+       let REPEATS=$REPEATS-1
+    done
+}
+
+# try and mount drive for up to 10 seconds or until success 
+_dir_wait 20 500 $LOGDIR
+
+if [ ! -x $LOGDIR ]; then
+  echo "No log directory found on SD Card"
+  sleep 1
+  exit
+fi
+
+echo "Log Directory Found on SD Card; logs will be periodically copied." | sbp_log $LOGLEVEL
+echo "Log Directory Found on SD Card; logs will be periodically copied."
+
+N=0
+while [[ -d $LOGDIR/$N ]] ; do
+    N=$(($N+1))
+done
+
+mkdir $LOGDIR/$N
+
+while true; do
+          #cp -R /var/log/* $LOGDIR/$N
+          rsync -r /var/log/ $LOGDIR/$N
```

Comment:
If there's not a huge speed difference between rsync and cp I would fall back to cp, adding rsync and dosfstools grows the image by .2 MB.  You can use cp -au (the -u only updates newer files) to accomplish much of the same thing without rsync (you'll just have to make sure the destination directory exists prior to copying, which is already done above), e.g:

```bash
cp -au /var/log/. $LOGDIR/$N/.
```

Or you could do a poor man's rsync with shell script like this:
```bash
sync()  {
  src=$1; shift
  dst=$1; shift
  while IFS= read -d $'\0' file; do
    dst_path="$dst/${file##$src/}"
    if [[ -d $file ]] && [[ ! -e $dst_path ]]; then
      mkdir -p $dst_path
      continue
    fi
    if [[ ! -e $dst_path ]]; then
      cp $file $dst_path
      continue
    fi
    if [[ $file -nt $dst_path ]]; then
      cp $file $dst_path
      continue
    fi
    echo "Not copying '$file' because it alreday exists and is not newer"
  done < <(find $src -mindepth 1 -print0)
}
```